client/components/Nav.tsx

```diff
@@ -1,16 +1,167 @@
-import React from "react";
-import { NavLink } from "react-router-dom";
+import React, {useState} from "react";
+import { NavLink, useNavigate} from "react-router-dom";
+import AppBar from "@mui/material/AppBar";
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import Box from '@mui/material/Box';
+import MenuIcon from '@mui/icons-material/Menu';
+import AccountCircle from '@mui/icons-material/AccountCircle';
+import SendIcon from '@mui/icons-material/Send'
+import MenuItem from '@mui/material/MenuItem';
+import Menu from '@mui/material/Menu';
+import { selectCurrentUser } from "../features/authSlice";
 
-type NavProps = {}
 
-const Nav = (props: NavProps) =>  {
-    return(
-        <nav>
+const Nav = () =>  {
+    const [menuIconAnchorEl, setMenuIconAnchorEl] = useState<null | HTMLElement>(null);
+    const [accountCircleAnchorEl, setAccountCircleAnchorEl] = useState<null | HTMLElement>(null);
+
+    const navigate = useNavigate();
+    const isLoggedIn = !!selectCurrentUser;
+
+    return (
+        <>
+            <AppBar 
+                position="static"
+            >
+                <Toolbar
+                    sx={{
+                        display: "flex", 
+                        justifyContent: "space-between",
+                        flexDirection: "row",
+                        flexWrap: "nowrap"
+                    }}
+                >
+                    <IconButton
+                        size="large"
+                        edge="start"
+                        color="inherit"
+                        aria-label="menu"
+                        sx={{ mr: 2 }}
+                        aria-controls="menu-appbar"
+                        aria-haspopup="true"
+                        onClick={e => setMenuIconAnchorEl(e.currentTarget)}
+                    >
+                        <MenuIcon />                
+                    </IconButton>
+                    <Menu
+                        anchorEl={menuIconAnchorEl}
+                        anchorOrigin={{
+                            vertical: 'top',
+                            horizontal: 'right',
+                        }}
+                        keepMounted
+                        transformOrigin={{
+                            vertical: 'top',
+                            horizontal: 'right',
+                        }}
+                        open={Boolean(menuIconAnchorEl)}
+                        onClose={e => setMenuIconAnchorEl(null)}
+                    >
+                        <MenuItem 
+                            onClick={e => {
+                                setMenuIconAnchorEl(null);
+                                navigate("/");
+                            }}
+                        > 
+                            Home
```

Comment:
tbh I tend to recommend instead of having an explicit link to home, that you just link to home when the header is clicked. It's a common design pattern on websites, and calling it Home in a user-facing link like this feels a bit old-fashioned to me